lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -134,6 +136,25 @@ export abstract class ClientApplication {
         return silentFlowClient.acquireToken(this.initializeRequestScopes(request) as SilentFlowRequest);
     }
 
+    /**
+     * Acquires tokens with password grant by exchanging client applications username and password for credentials
+     *
+     * This API is supported only for testing;
+     * The latest OAuth 2.0 Security Best Current Practice disallows the password grant entirely.
+     * This flow is added *only* for internal testing.
+     * @param request
+     */
+    async acquireTokenByUsernamePassword(request: UsernamePasswordRequest): Promise<AuthenticationResult> {
+        const usernamePasswordClientConfig = await this.buildOauthClientConfiguration(
+            request.authority
+        );
+        const usernamePasswordClient = new UsernamePasswordClient(
+            usernamePasswordClientConfig
+        );
+
+        return usernamePasswordClient.acquireToken(this.initializeRequestScopes(request) as UsernamePasswordRequest);
+    }
+
```

Comment:
This should go in PublicClientApplication. 

Also we should get rid of "this API is supported only for testing" and "This flow is added *only* for internal testing", and instead add something along the lines of "Using this API is not recommended", and possibly link to this which provides reasons why and workarounds: https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-acquire-token?tabs=java#this-flow-isnt-recommended

Reply:
Addressed the comments. I found relevant docs in Microsoft Identity, will also add a page in our repo.